src/libertem/udf/stddev.py

```diff
@@ -1,56 +1,133 @@
-import collections
-
 import numpy as np
+import numba
 
 from libertem.udf import UDF
 
 
-VariancePart = collections.namedtuple('VariancePart', ['var', 'sum_im', 'N'])
+@numba.njit(fastmath=True)
+def _merge(N, N0, sum_im0, var0, N1, sum_im1, var1):
+    # FIXME manual citation due to issues in CI.
+    # Check if :cite:`Schubert2018` works in a future release
+    '''
+    Basic function to perform numerically stable merge
+
+    Erich Schubert and Michael Gertz. Numerically stable parallel computation of
+    (co-)variance. In `Proceedings of the 30th International Conference on
+    Scientific and Statistical Database Management - SSDBM 18`. ACM Press, 2018.
+    `doi:10.1145/3221269.3223036 <https://doi.org/10.1145/3221269.3223036>`_
+
+    cite:Schubert2018
+    '''
+    # compute mean for each partitions
+    mean_A = sum_im0 / N0
+    mean_B = sum_im1 / N1
+
+    # compute mean for joint samples
+    delta = mean_B - mean_A
+    mean = mean_A + (N1 * delta) / N
+
+    # compute sum of images for joint samples
+    sum_im_AB = sum_im0 + sum_im1
+
+    # compute sum of variances for joint samples
+    delta_P = mean_B - mean
+    var_AB = var0 + var1 + (N1 * delta * delta_P)
+    return sum_im_AB, var_AB
 
 
-def merge(p0, p1):
+@numba.njit(fastmath=True)
+def merge(dest_N, dest_sum, dest_varsum, src_N, src_sum, src_varsum):
+    # FIXME manual citation due to issues in CI.
+    # Check if :cite:`Schubert2018` works in a future release
     """
     Given two sets of partitions, with sum of frames
-    and sum of variances, compute joint sum of frames
-    and sum of variances using one pass algorithm
+    and sum of variances, aggregate joint sum of frames
+    and sum of variances in destination partition using one pass
+    algorithm.
+
+    Erich Schubert and Michael Gertz. Numerically stable parallel computation of
+    (co-)variance. In `Proceedings of the 30th International Conference on
+    Scientific and Statistical Database Management - SSDBM 18`. ACM Press, 2018.
+    `doi:10.1145/3221269.3223036 <https://doi.org/10.1145/3221269.3223036>`_
+
+    cite:Schubert2018
 
     Parameters
     ----------
-    p0
-        Contains information about the first partition, including
-        sum of variances, sum of pixels, and number of frames used
+    dest_N : int
+        Number of frames aggregated in dest_sum and dest_varsum
+        The case :code:`dest_N == 0` is handled correctly.
+    dest_sum, dest_varsum
+        Aggregation buffers, will be updated: sum of pixels, sum of variances
+    src_N, src_sum, src_varsum
+        Information about the second partition
```

Comment:
Hmm, so where is the first partition? Maybe one should describe this as merging a "new" or "additional" partition into an intermediate result in the aggregation buffers?